independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/interceptors/producer/ProducerWithAbsractClassAndInterfaceInterceptionTest.java

```diff
@@ -0,0 +1,128 @@
+package io.quarkus.arc.test.interceptors.producer;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import jakarta.annotation.Priority;
+import jakarta.enterprise.context.Dependent;
+import jakarta.enterprise.inject.Produces;
+import jakarta.interceptor.AroundInvoke;
+import jakarta.interceptor.Interceptor;
+import jakarta.interceptor.InterceptorBinding;
+import jakarta.interceptor.InvocationContext;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.InterceptionProxy;
+import io.quarkus.arc.NoClassInterceptors;
+import io.quarkus.arc.test.ArcTestContainer;
+
+public class ProducerWithAbsractClassAndInterfaceInterceptionTest {
+    @RegisterExtension
+    public ArcTestContainer container = new ArcTestContainer(MyBinding1.class, MyInterceptor1.class,
+            MyBinding2.class, MyInterceptor2.class, MyProducer.class);
+
+    @Test
+    public void test() {
+        MyNonbeanBase nonbean = Arc.container().instance(MyNonbeanBase.class).get();
+        assertEquals("intercepted1: hello1_foobar", nonbean.hello1());
+        assertEquals("intercepted1: hello2_foobar", nonbean.hello2());
+        assertEquals("hello3", nonbean.hello3());
+        assertEquals("hello4_foobar", nonbean.hello4());
+        assertEquals("hello5", nonbean.hello5());
+    }
+
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target({ ElementType.TYPE, ElementType.METHOD, ElementType.CONSTRUCTOR })
+    @InterceptorBinding
+    @interface MyBinding1 {
+    }
+
+    @MyBinding1
+    @Priority(1)
+    @Interceptor
+    static class MyInterceptor1 {
+        @AroundInvoke
+        Object intercept(InvocationContext ctx) throws Exception {
+            return "intercepted1: " + ctx.proceed();
+        }
+    }
+
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target({ ElementType.TYPE, ElementType.METHOD, ElementType.CONSTRUCTOR })
+    @InterceptorBinding
+    @interface MyBinding2 {
+    }
+
+    @MyBinding2
+    @Priority(2)
+    @Interceptor
+    static class MyInterceptor2 {
+        @AroundInvoke
+        Object intercept(InvocationContext ctx) throws Exception {
+            return "intercepted2: " + ctx.proceed();
+        }
+    }
+
+    interface MyNonbean {
+        String hello1();
+
+        @MyBinding2 // this should be ignored, only class-level interceptors apply to implemented methods
+        String hello2();
+
+        @MyBinding2 // this should be ignored, only class-level interceptors apply to default methods
```

Comment:
I'm not entirely sure about this comment. The method is not overridden, so the binding might apply, but it's on an interface, so ... I'll have to take a proper look.

Reply:
It feels like `@NoClassInterceptors` should also be ignored on interfaces, but it isn't for now and that's a different issue. Shouldn't affect this PR.